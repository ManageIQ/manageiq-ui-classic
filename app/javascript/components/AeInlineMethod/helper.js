export const namespaceUrls = {
  aeMethodsUrl: '/miq_ae_class/ae_methods',
  aeMethodOperationsUrl: '/miq_ae_class/ae_method_operations',
  aeDomainsUrl: '/miq_ae_class/ae_domains',
};

export const noSelect = 'NONE';

/** Headers needed for the data-table list. */
export const methodSelectorHeaders = [
  {
    key: 'name',
    header: 'Name',
  },
  {
    key: 'path',
    header: 'Relative path',
  },
];

export const methodListHeaders = [
  ...methodSelectorHeaders,
  { key: 'remove', header: __('Remove'), actionCell: true },
];

/** Function to format the method data needed for the data-table list. */
export const formatMethods = (methods) => (methods.map((item) => ({
  id: item.id.toString(),
  name: { text: item.name, icon: 'icon node-icon fa-ruby' },
  path: item.relative_path,
})));

const removeMethodButton = () => ({
  is_button: true,
  actionCell: true,
  title: __('Remove'),
  text: __('Remove'),
  alt: __('Remove'),
  kind: 'danger',
  callback: 'removeMethod',
});

export const formatListMethods = (methods) => (methods.map((item, index) => ({
  id: item.id.toString(),
  name: { text: item.name, icon: 'icon node-icon fa-ruby' },
  path: item.relative_path,
  remove: removeMethodButton(item, index),
})));

/** Function to return a conditional URL based on the selected filters. */
export const searchUrl = (selectedDomain, text) => {
  const queryParams = [];
  if (selectedDomain && selectedDomain !== noSelect) {
    queryParams.push(`domain_id=${selectedDomain}`);
  }
  if (text && text !== noSelect) {
    queryParams.push(`search=${text}`);
  }
  const queryString = queryParams.length > 0 ? `?${queryParams.join('&')}` : '';
  return `${namespaceUrls.aeMethodsUrl}${queryString}`;
};
