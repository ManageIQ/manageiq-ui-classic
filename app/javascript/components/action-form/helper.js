const dataHelper = (values) => {
  const vd = values.description;
  const vt = values.action_type;
  const optionValue = values.options ? values.options : {};
  delete values.action_type;
  delete values.description;
  let data = {};
  switch (vt) {
    case 'delete_snapshots_by_age': {
      const optionsValue = parseInt(optionValue.age, 10);
      data = {
        name: vd,
        description: vd,
        action_type: vt,
        options: { age: optionsValue },
      };
      break;
    }
    case 'custom_automation': {
      const catValue = optionValue.ae_hash ? optionValue.ae_hash : [];
      const buildCatValue = (catValue) => {
        const obj = {};
        catValue.forEach((pt) => {
          obj[pt.attribute] = pt.value;
        });
        return obj;
      };
      const aeHash = buildCatValue(catValue);
      data = {
        name: vd,
        description: vd,
        action_type: vt,
        options: { ae_message: optionValue.ae_message, ae_request: optionValue.ae_request, ae_hash: aeHash },
      };
      break;
    }
    case 'inherit_parent_tags':
    case 'remove_tags': {
      const tagValue = optionValue.cats;
      const buildCatValue = (tagValue) => {
        const obj = [];
        tagValue.forEach((tg) => {
          obj.push(tg.value);
        });
        return obj;
      };
      const catOptions = buildCatValue(tagValue);
      data = {
        name: vd,
        description: vd,
        action_type: vt,
        options: { parent_type: optionValue.parent_type, cats: catOptions },
      };
      break;
    }
    case 'run_ansible_playbook': {
      const inventory = optionValue.inventory_type;
      if (inventory === 'LocalHost') {
        data = {
          name: vd,
          description: vd,
          action_type: vt,
          options: { service_template_name: [optionValue.service_template_name], use_localhost: true },
        };
      } else if (inventory === 'Target Machine') {
        data = {
          name: vd,
          description: vd,
          action_type: vt,
          options: { service_template_name: [optionValue.service_template_name], use_event_target: true },
        };
      } else {
        data = {
          name: vd,
          description: vd,
          action_type: vt,
          options: { service_template_name: [optionValue.service_template_name], hosts: optionValue.hosts },
        };
      }
      break;
    }
    case 'tag':
      data = {
        name: vd,
        description: vd,
        action_type: vt,
        options: { tags: [`/managed/${optionValue.tags}`] },
      };
      break;
    default:
      data = {
        name: vd,
        description: vd,
        action_type: vt,
        options: optionValue,
      };
  }
  return data;
};

const constructAeHash = (aeHashPairs) => {
  const options = [];
  const aeHashAttribute = Object.keys(aeHashPairs);
  aeHashAttribute.forEach((pt) => {
    const tempObj = { attribute: pt, value: aeHashPairs[pt] };
    options.push(tempObj);
  });
  return options;
};

const assignProfiles = [
  { label: __('Copy of sample'), value: 'Copy of sample' },
  { label: __('default'), value: 'default' },
  { label: __('host default'), value: 'host default' },
  { label: __('host sample'), value: 'host sample' },
  { label: __('sample'), value: 'sample' },
];

const findInitialValue = (inventoryType, recordId) => {
  let value = '';
  if (recordId === false || inventoryType === 'localhost') {
    value = 'LocalHost';
  } else if (inventoryType === 'event_target') {
    value = 'Target Machine';
  } else {
    value = 'Specific Hosts';
  }
  return value;
};

const buildParentTypeOptions = (parentType) => {
  const parentTypeArray = [];
  parentType.forEach((pt) => {
    const tempObj = { label: pt[0], value: pt[1] };
    parentTypeArray.push(tempObj);
  });
  parentTypeArray.splice(0, 1);
  return parentTypeArray;
};

const buildSnapShotAgeOptions = (snapshotAge) => {
  const snapshotAgeArray = [];
  snapshotAge.forEach((sa) => {
    const tempObj = { label: sa[0].toString(), value: sa[1].toString() };
    snapshotAgeArray.push(tempObj);
  });
  return snapshotAgeArray;
};

const buildInheritTagOptions = (inheritTags) => {
  const inheritTagArray = [];
  inheritTags.forEach((it) => {
    const tempObj = { label: it[1], value: it[0] };
    inheritTagArray.push(tempObj);
  });
  return inheritTagArray;
};

const buildAlertOptions = (alertOptions) => {
  const alertOptionArray = [];
  alertOptions.forEach((ao) => {
    const tempObj = { label: ao[0], value: ao[1] };
    alertOptionArray.push(tempObj);
  });
  return alertOptionArray;
};

const buildRunAnsible = (ansibleOptions) => {
  const ansibleOptionsArray = [];
  ansibleOptions.forEach((ao) => {
    const tempObj = { label: ao[0], value: ao[0] };
    ansibleOptionsArray.push(tempObj);
  });
  return ansibleOptionsArray;
};

const findLabel = (inheritTags, catLabel) => {
  let ansibleOptionsArray = [];
  inheritTags.forEach((pt) => {
    const tempObj = { label: pt[0], value: pt[1] };
    if (tempObj.value === catLabel) {
      ansibleOptionsArray = tempObj.label;
    }
  });
  return ansibleOptionsArray;
};

const buildTags = (tags, inheritTags) => {
  const entry = Object.entries(tags);
  const tagArray = [{ label: __('<Choose>'), value: '' }];
  entry.forEach((pt) => {
    const catOptions = [];
    const tempObj = { label: pt[0], value: pt[1] };
    tempObj.value.forEach((to) => {
      const val = findLabel(inheritTags, tempObj.label);
      const catObj = { label: to[0], value: `${val}/${to[0]}` };
      catOptions.push(catObj);
    });
    const tagObj = { label: tempObj.label.toUpperCase(), options: catOptions };
    tagArray.push(tagObj);
  });
  return tagArray;
};

export {
  dataHelper, constructAeHash, assignProfiles, findInitialValue, buildParentTypeOptions, buildSnapShotAgeOptions,
  buildInheritTagOptions, buildAlertOptions, buildRunAnsible, findLabel, buildTags,
};
