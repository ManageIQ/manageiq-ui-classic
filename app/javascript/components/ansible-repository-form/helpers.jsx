const GitUrlParse = require('git-url-parse');

export const urlValidation = (values) => {
  const errors = {};
  var error = false;
  try {
    const parsed = values.scm_url ? GitUrlParse(values.scm_url) : {};
    if ((parsed.protocol !== 'http'
          && parsed.protocol !== 'https'
          && parsed.protocol !== 'file'
          && parsed.protocol !== 'ssh')
          || parsed.pathname === '') {
      error = true;
    }
    if (parsed.protocols.length > 2 || (parsed.protocols.length === 1 &&  parsed.protocols[0] !== parsed.protocol)) {
      error = true;
    }
  } catch (err) {
    error = true;
  }
  if (error) {
    errors.scm_url = __('URL must include a protocol (http://, https:// or file://) with path or be a valid SSH path (user@server:path or ssh://user@address:port/path)');
  }
  return errors;
};
