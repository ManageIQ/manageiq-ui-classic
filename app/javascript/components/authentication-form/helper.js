export const getTimeOptions = (numOptions, multiplier) => {
  const options = [];
  const temp = [...Array(numOptions).keys()];
  temp.forEach((timeOption) => {
    options.push({ value: (timeOption * multiplier).toString(), label: (timeOption * multiplier).toString() });
  });
  return options;
};

export const secretKeyPlaceholder = () => {
  let placeholder = '';
  const length = [...Array(8).keys()];
  length.forEach(() => {
    placeholder += '\u25cf';
  });
  return placeholder;
};
