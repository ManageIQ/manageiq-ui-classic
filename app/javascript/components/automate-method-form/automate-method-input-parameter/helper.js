/** Action buttons for the input parameter table  */
export const InputParameterButtonActions = {
  EDIT: 'editInputParameter',
  DELETE: 'deleteInputParameter',
};

export const InputParameterRecordActions = {
  OPEN: 'openModal',
  CLOSE: 'closeModal',
  ADD: 'add',
  UPDATE: 'update',
  DELETE: 'delete',
};

const editInputParameterButton = () => ({
  is_button: true,
  title: __('Edit'),
  text: __('Edit'),
  alt: __('Edit'),
  kind: 'ghost',
  callback: InputParameterButtonActions.EDIT,
});

const deleteInputParameterButton = () => ({
  is_button: true,
  title: __('Delete'),
  text: __('Delete'),
  alt: __('Delete'),
  kind: 'ghost',
  callback: InputParameterButtonActions.DELETE,
});

/** Input parameter data for table  */
export const reformatList = (items) => items.map((item, index) => ({
  ...item,
  id: index.toString(),
  edit: editInputParameterButton(item, index),
  delete: deleteInputParameterButton(item, index),
}));

export const headers = [
  { key: 'inputName', header: __('Input Name') },
  { key: 'dataType', header: __('Data Type') },
  { key: 'defaultValue', header: __('Default Value') },
  { key: 'edit', header: __('Edit') },
  { key: 'delete', header: __('Delete') },
];

/* Function to handle the action buttons  */
export const handleInputParameterUpdate = (actionType, data, formData) => {
  const { inputParameter } = formData;

  if (actionType !== InputParameterRecordActions.DELETE) {
    inputParameter.modal = false;
  }

  switch (actionType) {
    case InputParameterRecordActions.OPEN:
      inputParameter.modal = true;
      if (data && data.selectedId) {
        inputParameter.selectedId = data.selectedId;
      }
      break;
    case InputParameterRecordActions.CLOSE:
      inputParameter.modal = false;
      inputParameter.selectedId = undefined;
      break;
    case InputParameterRecordActions.ADD:
      inputParameter.items.push(data.values);
      inputParameter.selectedId = undefined;
      break;
    case InputParameterRecordActions.UPDATE:
      inputParameter.items[inputParameter.selectedId] = data.values;
      inputParameter.selectedId = undefined;
      break;
    case InputParameterRecordActions.DELETE:
      inputParameter.items.splice(data.selectedId, 1);
      inputParameter.selectedId = undefined;
      break;
    default:
      console.warn(__('Unknown action'));
  }

  return { ...formData.inputParameter };
};

/** Helper function to get provider details and restructure its options */
export const initialState = {
  manager_id: null,
};

export const reducer = (state, action) => {
  switch (action.type) {
    case 'SET_MANAGER_ID':
      return {
        ...state,
        manager_id: action.payload,
      };
    default:
      return state;
  }
};
