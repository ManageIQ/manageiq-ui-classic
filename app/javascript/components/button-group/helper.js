/** Function to change the format of the props available_fields & fields from the format
   * [[string,number]] to [{label:string, value:number}] to use as options in dual-list-select
   * available_fields or an empty array is passed as props. */
export const formatButton = (buttons, isGenericObject = false) => {
  if (buttons.length <= 0) {
    return [];
  }
  const options = [];
  if (isGenericObject) {
    buttons.forEach((btn) => options.push({ label: btn.name, value: parseInt(btn.id, 10) }));
  } else {
    buttons.forEach((btn) => options.push({ label: btn[0], value: btn[1] }));
  }
  return options;
};

/** Function to extract the ButtonGroupName from api result like 'ButtonGroupName | xyz'. */
export const formatName = (buttonGroupName) => {
  if (!buttonGroupName) {
    return undefined;
  }
  return buttonGroupName.split('|')[0].trim('');
};

/** Function to update the field set_data's values during form submit */
export const formatSetData = (setData, buttonIcon, appliesToClass, appliesToId) => {
  if (appliesToId) {
    return ({
      ...setData,
      button_color: (setData && setData.button_color) ? setData.button_color : '#000',
      button_icon: buttonIcon,
      applies_to_class: appliesToClass,
      applies_to_id: appliesToId,
    });
  }
  return ({
    ...setData,
    button_color: (setData && setData.button_color) ? setData.button_color : '#000',
    button_icon: buttonIcon,
    applies_to_class: appliesToClass,
  });
};

export const getGenericObjectButtonList = (buttonGroups, data, recId) => {
  const buttons = [];
  const assignedButtons = [];
  let unassignedButtons = [];

  if (buttonGroups.resources) {
    buttonGroups.resources.forEach((buttonGroup) => {
      if (buttonGroup.set_data.button_order) {
        if (recId !== parseInt(buttonGroup.id, 10)) {
          assignedButtons.push(...buttonGroup.set_data.button_order);
        }
      }
    });
  }

  if (data.resources) {
    data.resources.forEach((button) => {
      unassignedButtons.push(parseInt(button.id, 10));
    });

    unassignedButtons = unassignedButtons.filter((button) => !assignedButtons.includes(button));

    data.resources.forEach((button) => {
      if (unassignedButtons.includes(parseInt(button.id, 10))) {
        buttons.push({ label: button.name, value: parseInt(button.id, 10) });
      }
    });
  }

  return buttons;
};
