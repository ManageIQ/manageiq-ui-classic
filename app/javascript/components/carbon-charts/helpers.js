export const sampleData = [{
  group: 'Dataset 1',
  key: '1',
  value: 100,
},
{
  group: 'Dataset 1',
  key: '2',
  value: 400,
},
{
  group: 'Dataset 1',
  key: '4',
  value: 500,
},
{
  group: 'Dataset 1',
  key: '3',
  value: 200,
},
{
  group: 'Dataset 1',
  key: '5',
  value: 150,
},
{
  group: 'Dataset 2',
  key: '1',
  value: 280,
},
{
  group: 'Dataset 2',
  key: '2',
  value: 350,
},
{
  group: 'Dataset 2',
  key: '3',
  value: 700,
},
{
  group: 'Dataset 2',
  key: '5',
  value: 600,
},
{
  group: 'Dataset 2',
  key: '5',
  value: 200,
},
{
  group: 'Dataset 3',
  key: '4',
  value: 100,
},
{
  group: 'Dataset 3',
  key: '2',
  value: 500,
},
{
  group: 'Dataset 3',
  key: '1',
  value: 200,
}];

export const pieData = [{
  group: 'Dataset 1',
  key: '1',
  value: 100,
},
{
  group: 'Dataset 2',
  key: '1',
  value: 280,
},
{
  group: 'Dataset 3',
  key: '4',
  value: 100,
},
{
  group: 'Dataset 4',
  key: '2',
  value: 500,
}];

// convert report data to carbon chart data format.
export const getConvertedData = (data) => {
  if (data && data.data.columns && data.data && data.miq && data.miq.category_table && data.miq.name_table) {
    const columnsData = data.data.columns;
    const dataGroups = data.miq.category_table;
    const rowsData = data.miq.name_table;
    const arr = [];
    if (columnsData.length > 0) {
      columnsData.forEach((items) => {
        items.forEach((item, i) => {
          const obj = {};
          if (i !== 0 && rowsData[items[0]]) {
            obj.group = rowsData[items[0]];
            obj.key = dataGroups[i - 1];
            obj.value = item;
            arr.push(obj);
          }
        });
      });
    }
    return arr;
  }
  return [];
};

export const getLineConvertedData = (data) => {
  const columnsData = data.data.columns;
  const dataGroups = data.miq.category_table;
  const rowsData = data.miq.name_table;
  const arr = [];
  columnsData.forEach((items) => {
    items.forEach((item, i) => {
      const obj = {};
      if (i !== 0 && rowsData[items[0]]) {
        obj.group = rowsData[items[0]];
        obj.key = new Date(dataGroups[i - 1]);
        obj.value = item;
        arr.push(obj);
      }
    });
  });
  return arr;
};
