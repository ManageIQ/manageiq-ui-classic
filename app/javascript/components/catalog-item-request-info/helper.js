// const serviceSelectedVms = {
//   label: __('Selected VM'),
//   keys: ['src_vm_id', 'provision_type', 'linked_clone', 'seal_template'],
// };

// const serviceSelect = {
//   label: __('Select'),
//   keys: ['vm_filter', 'src_vm_id', 'provision_type', 'linked_clone', 'snapshot'],
// };

// const serviceSupportsPxe = {
//   label: __('PXE'),
//   keys: ['pxe_server_id', 'pxe_image_id', 'windows_image_id'],
// };

// const serviceSupportsIso = {
//   label: __('ISO'),
//   keys: ['iso_image_id'],
// };

// const serviceCloudManager = (cloudManager) => ({
//   label: cloudManager ? __('Number of Instances') : __('Number of VMs'),
//   keys: ['number_of_vms'],
// });

// const serviceNaming = {
//   label: __('Naming'),
//   keys: ['vm_name', 'vm_description', 'vm_prefix'],
// };

// export const serviceKeys = (item) => {
//   const serviceData = [];
//   serviceData.push(item.isOvirtInfraManager ? serviceSelectedVms : serviceSelect);
//   if (item.supportsPxe) serviceData.push(serviceSupportsPxe);
//   if (item.supportsIso) serviceData.push(serviceSupportsIso);
//   serviceData.push(serviceCloudManager(item.cloudManager));
//   serviceData.push(serviceNaming);
//   return serviceData;
// };
