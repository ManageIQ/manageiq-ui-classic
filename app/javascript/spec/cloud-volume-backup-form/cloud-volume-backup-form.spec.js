import React from 'react';
import toJson from 'enzyme-to-json';
import fetchMock from 'fetch-mock';
import { act } from 'react-dom/test-utils';
import { mount } from '../helpers/mountForm';
import CloudVolumeBackupForm from '../../components/cloud-volume-backup-form';
import { restoreData } from '../../components/cloud-volume-backup-form/helper';

require('../helpers/miqSparkle');
require('../helpers/miqAjaxButton');

const cloudVolume = { recordId: '23', name: 'Cloud volume name', type: 'restore' };

describe('Cloud Volume backup form component', () => {
  beforeEach(() => {
    fetchMock
      .mock(`/api/cloud_volumes?expand=resources&attributes=name,id`, {});
  });
  afterEach(() => {
    fetchMock.reset();
    fetchMock.restore();
  });

  const data = restoreData(cloudVolume.recordId, cloudVolume.name, cloudVolume.type);

  it('should render the restore cloud volume backup form', () => {
    const wrapper = mount(<CloudVolumeBackupForm
      recordId={cloudVolume.recordId}
      name={cloudVolume.name}
      type={cloudVolume.type}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('when restoring cloud volume from backup', async(done) => {
    // fetchMock.getOnce('/api/cloud_volumes?expand=resources&attributes=name,id', {});
    const payload = {
      volume: { id: 100, name: 'name' },
      button: 'restore',
      id: 1,
      name: '',
    };
    fetchMock.postOnce(data.save.postUrl, payload);
    let wrapper;
    await act(async() => {
      wrapper = mount(<CloudVolumeBackupForm
        recordId={cloudVolume.recordId}
        name={cloudVolume.name}
        type={cloudVolume.type}
      />);
    });
    wrapper.update();
    expect(toJson(wrapper)).toMatchSnapshot();
    done();
  });
});
