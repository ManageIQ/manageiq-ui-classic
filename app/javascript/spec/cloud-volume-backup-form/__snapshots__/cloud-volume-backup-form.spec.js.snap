// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cloud Volume backup form component should render the restore cloud volume backup form 1`] = `
<Provider
  store={
    Object {
      "asyncReducers": Object {
        "FormButtons": [Function],
        "notificationReducer": [Function],
      },
      "dispatch": [Function],
      "getState": [Function],
      "injectReducers": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <CloudVolumeBackupForm
    name="Cloud volume name"
    recordId="23"
    type="restore"
  />
</Provider>
`;

exports[`Cloud Volume backup form component when restoring cloud volume from backup 1`] = `
<Provider
  store={
    Object {
      "asyncReducers": Object {
        "FormButtons": [Function],
        "notificationReducer": [Function],
      },
      "dispatch": [Function],
      "getState": [Function],
      "injectReducers": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <CloudVolumeBackupForm
    name="Cloud volume name"
    recordId="23"
    type="restore"
  />
</Provider>
`;
